app/src/main/java/com/example/vaccinespotter/notifications/VaccineNotificationManager.java

```diff
@@ -13,24 +13,32 @@
 
 public class VaccineNotificationManager {
 
+    private final static int MAX_CHARACTERS_EXCEPTION_STRING = 100;
     private final String VACCINE_AVAILABLE = "Vaccine available";
     private final String TAG = "VaccineNotificationManager";
     private int mNotificationId;
     private final Context mContext;
     private List<NotificationModel> sentNotifications;
+    private boolean mRegisteredNotification = false;
 
     public VaccineNotificationManager(Context context) {
         mContext = context;
-        sentNotifications = new ArrayList<NotificationModel>();
+        sentNotifications = new ArrayList<>();
     }
 
     public void registerNotifications() {
-        NotificationHelper.createNotificationChannel(mContext,
-            NotificationManagerCompat.IMPORTANCE_DEFAULT, false,
-            mContext.getString(R.string.app_name), "Vaccine Spotter App notification channel.");
+        if (!mRegisteredNotification) {
+            NotificationHelper.createNotificationChannel(mContext,
+                NotificationManagerCompat.IMPORTANCE_DEFAULT, false,
+                mContext.getString(R.string.app_name), "Vaccine Spotter App notification channel.");
+            mRegisteredNotification = true;
+        }
     }
 
     public void showNotifications(List<NotificationModel> models) {
+        if (models == null)
```

Comment:
Redundant check as the collection would be empty is nothing is added?

Reply:
It can be null if an exception occurs in the queryCowin, if there is no exception it will be empty which won't be a problem for our code.
In case of null for each loop will crash.